packages/apollo-engine-reporting/src/__tests__/extension.test.ts

```diff
@@ -83,3 +83,67 @@ test('trace construction', async () => {
   requestDidEnd();
   // XXX actually write some tests
 });
+
+const variables: Record<string, any> = {
+  testing: 'testing',
+  t2: 2,
+};
+
+test('check variableJson output for privacyEnforcer boolean type', () => {
+  // Case 1: No keys/values in variables to be filtered/not filtered
+  const emptyOutput = new Trace.Details();
+  expect(makeTraceDetails({}, true)).toEqual(emptyOutput);
+  expect(makeTraceDetails({}, false)).toEqual(emptyOutput);
+
+  // Case 2: Filter all variables (enforce == True)
+  const filteredOutput = new Trace.Details();
+  Object.keys(variables).forEach(name => {
+    filteredOutput.variablesJson[name] = '';
+  });
+  expect(makeTraceDetails(variables, true)).toEqual(filteredOutput);
+
+  // Case 3: Do not filter variables (enforce == False)
+  const nonFilteredOutput = new Trace.Details();
+  Object.keys(variables).forEach(name => {
+    nonFilteredOutput.variablesJson[name] = JSON.stringify(variables[name]);
+  });
+  expect(makeTraceDetails(variables, false)).toEqual(nonFilteredOutput);
+});
+
+test('variableJson output for privacyEnforcer Array type', () => {
+  const privacyEnforcerArray: string[] = ['testing', 'notInVariables'];
+  const expectedVariablesJson = {
+    testing: '',
+    t2: JSON.stringify(2),
+  };
+  expect(
+    makeTraceDetails(variables, privacyEnforcerArray).variablesJson,
+  ).toEqual(expectedVariablesJson);
+});
+
+test('variableJson output for privacyEnforcer custom function', () => {
+  // Custom function that redacts every variable to 100;
+  const modifiedValue = 100;
+  const customEnforcer = (input: Record<string, any>): Record<string, any> => {
+    let out: Record<string, any> = {};
+    Object.keys(input).map((name: string) => {
+      out[name] = modifiedValue;
+    });
+    return out;
+  };
+
+  // Expected output
+  const output = new Trace.Details();
```

Comment:
Are you still planning to include a new boolean that means "a custom function was applied"?

Reply:
I decided to save these changes for another PR, because that would require some protobuf changes in both monorepo + apollo-server, and other edits the frontend repo as well.

Reply:
Reasonable choice! Maybe leave a TODO comment?